### ISSUE TITLE
Parallel implementation [question]

### ISSUE THREAD

**Issue by Dan-Treacher** [wwwjk366/gower#5]
Hi there

I was wondering whether anyone had any insights into methods that could be used to parallelise the gower distance calculation for data with mixed numerical and nominal categorical.

I've seen some people suggesting using `sklearn.metrics.pairwise_distances` with a custom function for the `metric=` argument, but I think `pairwise_distances` can only take numerical inputs which wouldn't work with high cardinality nominal categorical data (one hot encoding would result in thousands of columns)

https://hal.archives-ouvertes.fr/hal-02047514/document This paper might start some discussion?

Thanks

**Comment by adinathauti** [comment]
Perhaps a similar type of request

Would it be possible to get a sklearn.neighbors.KDTree like structure for quick nearest neighbour queries?

The default sklearn KDTree does not allow for callable functions as metrics
